core/payment-driver/erc20/src/erc20/ethereum.rs

```diff
@@ -0,0 +1,287 @@
+    let node_id = NodeId::from(address.as_ref());
+    let signature = bus::sign(node_id, eth_utils::get_tx_hash(&tx, chain_id)).await?;
+
+    Ok(raw_tx_to_entity(
+        &tx,
+        address,
+        chain_id,
+        Utc::now(),
+        &signature,
+        TxType::Faucet,
+    ))
+}
+
+pub async fn sign_transfer_tx(
+    address: H160,
+    recipient: H160,
+    amount: U256,
+    network: Network,
+    nonce: U256,
+) -> Result<TransactionEntity, GenericError> {
+    let env = get_env(network);
+    let client = get_client(network)?;
+    let contract = prepare_erc20_contract(&client, &env)?;
+
+    let data = eth_utils::contract_encode(&contract, TRANSFER_ERC20_FUNCTION, (recipient, amount))
+        .map_err(GenericError::new)?;
+    let gas_price = client.eth().gas_price().await.map_err(GenericError::new)?;
+
+    let tx = RawTransaction {
+        nonce,
+        to: Some(contract.address()),
+        value: U256::from(0),
+        gas_price,
+        gas: *GLM_TRANSFER_GAS,
+        data,
+    };
+    let chain_id = get_chain_id(network);
+    let node_id = NodeId::from(address.as_ref());
+    let signature = bus::sign(node_id, eth_utils::get_tx_hash(&tx, chain_id)).await?;
+
+    Ok(raw_tx_to_entity(
+        &tx,
+        address,
+        chain_id,
+        Utc::now(),
+        &signature,
+        TxType::Faucet,
+    ))
+}
+
+pub async fn send_tx(signed_tx: Vec<u8>, network: Network) -> Result<H256, GenericError> {
+    let client = get_client(network)?;
+    let tx_hash = client
+        .eth()
+        .send_raw_transaction(Bytes::from(signed_tx))
+        .await
+        .map_err(GenericError::new)?;
+    Ok(tx_hash)
+}
+
+pub async fn is_tx_confirmed(
+    tx_hash: H256,
+    current_block: &U64,
+    network: Network,
+) -> Result<bool, GenericError> {
+    let env = get_env(network);
+    let tx = get_tx_receipt(tx_hash, network).await?;
+    if let Some(tx) = tx {
+        if let Some(b) = tx.block_number {
+            if b + env.required_confirmations >= *current_block {
+                return Ok(true);
+            }
+        }
+    }
+    Ok(false)
+}
+
+pub async fn get_tx_receipt(
+    tx_hash: H256,
+    network: Network,
+) -> Result<Option<TransactionReceipt>, GenericError> {
+    let client = get_client(network)?;
+    let result = client
+        .eth()
+        .transaction_receipt(tx_hash)
+        .await
+        .map_err(GenericError::new)?;
+    Ok(result)
+}
+
+fn get_rpc_addr_from_env(network: Network) -> String {
+    match network {
+        Network::Mainnet => std::env::var("ERC20_MAINNET_GETH_ADDR")
+            .unwrap_or("https://geth.golem.network:55555".to_string()),
+        Network::Rinkeby => std::env::var("ERC20_RINKEBY_GETH_ADDR")
+            .unwrap_or("http://geth.testnet.golem.network:55555".to_string()),
+    }
+}
+
+fn get_chain_id(network: Network) -> u64 {
+    match network {
+        Network::Mainnet => 1,
+        Network::Rinkeby => 4,
+    }
```

Comment:
This mapping is already defined in `Network` enum itself. I think `network as u64` should work here.

Reply:
fixed